cabal-install/Distribution/Client/RebuildMonad.hs

```diff
@@ -166,3 +176,58 @@ monitorDirectoryStatus dir = do
                     then monitorDirectory dir
                     else monitorNonExistentDirectory dir]
 
+doesFileExistMonitored :: FilePath -> Rebuild Bool
+doesFileExistMonitored f = do
+    root <- askRoot
+    exists <- liftIO $ doesFileExist (root </> f)
+    -- TODO: If the file exists, should we really monitor the entire
+    -- file?!
```

Comment:
The gist of this comment is different. `monitorFile` sets up a dependence on the timestamp of the file. But consider a build process which tests if a file exists, and does A if it does and B otherwise. We do NOT need to rebuild if the timestamp on the file updates. But there is no way to express this in the current MonitorFile data structure.